src/x/config/pooling.go

```diff
@@ -22,51 +22,31 @@ package config
 
 import "github.com/m3db/m3/src/x/sync"
 
-const (
-	defaultWorkerPoolStaticSize = 4096
```

Comment:
i found this default pretty strange. the only place it's ever used is the query read pool, since the query write pool explicitly sets a default value (which is also 4096)